src/main/java/com/team4/testingsystem/services/impl/RestrictionsServiceImpl.java

```diff
@@ -202,4 +207,19 @@ public void checkNotArchivedContentFile(ContentFile contentFile) {
             throw new QuestionOrTopicEditingException("You can't edit archived topics");
         }
     }
+
+    @Override
+    public void checkListeningHasAudio(ContentFile contentFile) {
+        if (contentFile.getUrl() == null) {
+            throw new NoAudioException("You can't add listening topics without audios");
+        }
+    }
+
+    @Override
+    public void checkExistsOnS3(String fileName) {
+        Path filePath = Path.of(FileUtils.getTempDirectory() + "/testing-system/" + fileName);
```

Comment:
I don't think this check is correct. Temp files aren't stored permanently, that's why they are temp :)

Reply:
Amazon S3 SDK has this method:
`boolean doesObjectExist(String bucketName, String objectName)`
I think that's what you need

Reply:
We have several filestorages (at least local one and s3). And they have different logic for such checks. I think we should add isFileExist method to the interface and implement it on all the file storages. Then we should call the method here and throw the exception if file is not exist